src/libraries/System.Drawing.Common/src/System/Drawing/DrawingComWrappers.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    /// <summary>
+    /// The ComWrappers implementation for System.Drawing.Common's COM interop usages.
+    ///
+    /// Supports IStream and IPicture COM interfaces.
+    /// </summary>
+    internal unsafe class DrawingComWrappers : ComWrappers
+    {
+        private static readonly ComInterfaceEntry* s_wrapperEntry = InitializeComInterfaceEntry();
+        private static readonly Lazy<DrawingComWrappers> s_instance = new Lazy<DrawingComWrappers>(() => new DrawingComWrappers());
+
+        private DrawingComWrappers() { }
+
+        internal static DrawingComWrappers Instance => s_instance.Value;
+
+        internal static void CheckStatus(int result)
+        {
+            if (result != 0)
+            {
+                throw new ExternalException() { HResult = result };
+            }
+        }
+
+        private static ComInterfaceEntry* InitializeComInterfaceEntry()
+        {
+            GetIUnknownImpl(out IntPtr fpQueryInteface, out IntPtr fpAddRef, out IntPtr fpRelease);
+
+            IStreamVtbl iStreamVtbl = IStreamVtbl.Create(fpQueryInteface, fpAddRef, fpRelease);
+
+            IntPtr iStreamVtblRaw = RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), sizeof(IStreamVtbl));
+            Marshal.StructureToPtr(iStreamVtbl, iStreamVtblRaw, false);
```

Comment:
This is costly. I would instead allocate the memory cast to the desired type and populate in `IStreamVtbl.Create()`.